src/ext/serial.cpp

```diff
@@ -65,24 +118,41 @@ template<typename T> class ThingSetSerial: public ThingSetStream
  */
 
 #ifdef UART_SERIAL_ENABLED
-    extern const int pub_channel_serial;
-    extern Serial serial;
 
-    ThingSetSerial ts_uart(serial, pub_channel_serial);
+    #ifdef __MBED__
+        extern const int pub_channel_serial;
+        extern Serial serial;
+        ThingSetSerial ts_uart(serial, pub_channel_serial);
+
+    #elif defined(__ZEPHYR__)
+        extern const int pub_channel_serial;
+        ThingSetSerial ts_uart(pub_channel_serial);
+    #endif /* MBED or ZEPHYR */
+
 #endif /* UART_SERIAL_ENABLED */
 
 // ToDo: Add USB serial again (previous library was broken with recent mbed releases)
 
-char ThingSetStream::buf_resp[1000];
-
 extern ThingSet ts;
 
+int len_ts;
+
 void ThingSetStream::process_1s()
 {
     if (ts.get_pub_channel(channel)->enabled) {
+        #ifdef __MBED__
         ts.pub_msg_json(buf_resp, sizeof(buf_resp), channel);
         stream->puts(buf_resp);
         stream->putc('\n');
+
+        #elif defined(__ZEPHYR__)
+        int len = ts.pub_msg_json(buf_resp, sizeof(buf_resp), channel);
```

Comment:
Please put outside of `#ifdef` to avoid duplicated code. (same for mbed and zephyr)